pkg/storage/etcd/etcd_watcher_secure_test.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	rt "runtime"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/testapi"
+	"k8s.io/kubernetes/pkg/storage"
+	"k8s.io/kubernetes/pkg/storage/etcd/etcdtest"
+	etcdtesting "k8s.io/kubernetes/pkg/storage/etcd/testing"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"golang.org/x/net/context"
+)
+
+func TestSecuredWatch(t *testing.T) {
```

Comment:
My point earlier was why not just default ```NewEtcdTestClientServer``` to always be secure, then you could eliminate the duplicate set of tests.  I can't think of a good reason not to default all tests to secure, and you would get far more coverage, to the point where I would not be concerned about integration.  In general we need to tackle integration at a later date to use the same infra.